src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseParticipationResource.java

```diff
@@ -242,8 +242,7 @@ public ResponseEntity<Map<Long, Optional<ProgrammingSubmission>>> getLatestPendi
      */
     @PutMapping("programming-exercise-participations/{participationId}/reset-repository")
     @EnforceAtLeastStudent
-    public ResponseEntity<Void> resetRepository(@PathVariable Long participationId, @RequestParam(required = false) Long gradedParticipationId)
-            throws GitAPIException, IOException {
+    public ResponseEntity<Void> resetRepository(@PathVariable Long participationId, @RequestParam Optional<Long> gradedParticipationId) throws GitAPIException, IOException {
```

Comment:
Here we would also need the required `false` flag
```suggestion
    public ResponseEntity<Void> resetRepository(@PathVariable Long participationId, @RequestParam(required = false) Optional<Long> gradedParticipationId) throws GitAPIException, IOException {
```

Reply:
see above